apps/builder/app/builder/features/settings-panel/controls/code.tsx

```diff
@@ -70,8 +70,17 @@ const validateHtml = (value: string): Error | undefined => {
   const div = document.createElement("div");
   div.innerHTML = value;
   const expected = div.innerHTML;
-  // We don't need to show error for unnecessary whitespace.
-  if (value.replace(/\s/g, "") !== expected.replace(/\s/g, "")) {
+  const clean = (value: string) => {
+    return (
```

Comment:
What's wrong with

```
function isValidHTML(htmlString) {
    const parser = new DOMParser();
    const doc = parser.parseFromString(htmlString, 'text/html');
    return !doc.querySelector('parsererror');
}
```

What the case of `clean(blabla) === clean(otherblabla)`

Reply:
I tried it, nothing was triggering a parseerror, it may be not implemented in chrome, so it's kinda useless, then you get the same issue I am fixing above

Reply:
The problem is to identify broken HTML and because browser is flexible and always tries to fix it we never know if something is really broken, so what I am trying to do there is to minimize the amount of false positives:
1. different amount of whitespace
2. `boolean=""` is the same as `boolean`

Reply:
lets add a comment then